packages/dev/serializers/src/glTF/2.0/glTFExporter.ts

```diff
@@ -422,6 +449,288 @@ export class _Exporter {
+            }
+        }
+    }
+
+    /**
+     * Reorders the vertex attribute data based on the primitive mode.  This is necessary when indices are not available and the winding order is
+     * clock-wise during export to glTF
+     * @param submesh BabylonJS submesh
+     * @param primitiveMode Primitive mode of the mesh
+     * @param vertexBufferKind The type of vertex attribute
+     * @param meshAttributeArray The vertex attribute data
+     * @param byteOffset The offset to the binary data
+     * @param binaryWriter The binary data for the glTF file
+     */
+    private _reorderVertexAttributeDataBasedOnPrimitiveMode(
+        submesh: SubMesh,
+        primitiveMode: number,
+        vertexBufferKind: string,
+        meshAttributeArray: FloatArray,
+        byteOffset: number,
+        binaryWriter: _BinaryWriter
+    ): void {
+        switch (primitiveMode) {
+            case Material.TriangleFillMode: {
+                this._reorderTriangleFillMode(submesh, vertexBufferKind, meshAttributeArray, byteOffset, binaryWriter);
+                break;
+            }
+            case Material.TriangleStripDrawMode: {
+                this._reorderTriangleStripDrawMode(submesh, vertexBufferKind, meshAttributeArray, byteOffset, binaryWriter);
+                break;
+            }
+            case Material.TriangleFanDrawMode: {
+                this._reorderTriangleFanMode(submesh, vertexBufferKind, meshAttributeArray, byteOffset, binaryWriter);
+                break;
+            }
+        }
+    }
+
+    /**
+     * Reorders the vertex attributes in the correct triangle mode order .  This is necessary when indices are not available and the winding order is
+     * clock-wise during export to glTF
+     * @param submesh BabylonJS submesh
+     * @param vertexBufferKind The type of vertex attribute
+     * @param meshAttributeArray The vertex attribute data
+     * @param byteOffset The offset to the binary data
+     * @param binaryWriter The binary data for the glTF file
+     */
+    private _reorderTriangleFillMode(submesh: SubMesh, vertexBufferKind: string, meshAttributeArray: FloatArray, byteOffset: number, binaryWriter: _BinaryWriter) {
+        const vertexBuffer = this._getVertexBufferFromMesh(vertexBufferKind, submesh.getMesh() as Mesh);
+        if (vertexBuffer) {
+            const stride = vertexBuffer.byteStride / VertexBuffer.GetTypeByteLength(vertexBuffer.type);
+            if (submesh.verticesCount % 3 !== 0) {
+                Tools.Error("The submesh vertices for the triangle fill mode is not divisible by 3!");
+            } else {
+                const vertexData: Vector2[] | Vector3[] | Vector4[] = [];
+                let index = 0;
+                switch (vertexBufferKind) {
+                    case VertexBuffer.PositionKind:
+                    case VertexBuffer.NormalKind: {
+                        for (let x = submesh.verticesStart; x < submesh.verticesStart + submesh.verticesCount; x = x + 3) {

```

Comment:
Is it not possible to factorize the implementation for all vertex buffer types by not creating arrays of `Vector2/3/4` but by cloning the `meshAttributeArray` and performing operations in this cloned array?

Reply:
I think you are reviewing existing code. I deleted this originally and had to revive it since it was still necessary.